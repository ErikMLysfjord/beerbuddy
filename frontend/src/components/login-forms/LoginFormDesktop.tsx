import styles from "./LoginFormDesktop.module.css";
import { ValidateErrorEntity } from "rc-field-form/lib/interface";
import { Card, Form, Input, Button, theme } from "antd";
import Logo from "../logo/Logo";

const { useToken } = theme;

type FieldType = {
  username?: string;
  password?: string;
};

interface LoginFormDesktopProps {
  showMessage: (string: { username: string; password: string }) => void;
  onFinishFailed: (errorInfo: ValidateErrorEntity) => void;
}

const LoginFormDesktop = ({
  showMessage,
  onFinishFailed,
}: LoginFormDesktopProps) => {
  const { token } = useToken();
  return (
    <div className={styles.container}>
      <div className={styles.logoContainer}>
        <Logo />
      </div>
      <div className={styles.loginContainer}>
        <div className={styles.loginText}>
          <h2>
            {"Taste, Rate, Repeat -"}
            <br /> {"The craft beer journey"}
          </h2>
          <p>
            {"Join the RateMyBrew community to unlock exclusive"}
            <br />
            {"features and be part of a global craft beer conversation."}
          </p>
          <a
            style={{ color: token.colorPrimary }}
            className={styles.loginLink}
            href="/"
          >
            {"Not a user? Sign up here"}
          </a>
        </div>
        <div className={styles.loginFormContainer}>
          <Card
            style={{
              height: "100%",
              backgroundColor: token.colorPrimaryBg,
            }}
            bodyStyle={{ padding: "20", height: "100%" }}
          >
            <h1>Log in</h1>
            <Form
              name="basic"
              className={styles.loginForm}
              layout="vertical"
              initialValues={{ remember: true }}
              onFinish={showMessage}
              onFinishFailed={onFinishFailed}
              autoComplete="off"
            >
              <Form.Item<FieldType>
                label="Username"
                name="username"
                rules={[
                  { required: true, message: "Please input your username!" },
                ]}
              >
                <Input />
              </Form.Item>

              <Form.Item<FieldType>
                label="Password"
                name="password"
                rules={[
                  { required: true, message: "Please input your password!" },
                ]}
              >
                <Input.Password className={styles.loginFormPasswordContainer} />
              </Form.Item>
              <Form.Item>
                <Button
                  type="primary"
                  htmlType="submit"
                  className={styles.loginButton}
                >
                  Submit
                </Button>
              </Form.Item>
            </Form>
          </Card>
        </div>
      </div>
    </div>
  );
};

export default LoginFormDesktop;
