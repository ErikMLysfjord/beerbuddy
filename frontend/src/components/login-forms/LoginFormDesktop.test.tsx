import { describe, it, expect, vi } from "vitest";
import { render } from "@testing-library/react";
import LoginFormDesktop from "./LoginFormDesktop";
import { axe } from "jest-axe";

const showMessage = vi.fn();
const onFinishFailed = vi.fn();
const saveUser = vi.fn();

describe("LoginFormDesktop", () => {
  it("is accessible", async () => {
    const { container } = render(
      <LoginFormDesktop
        showMessage={(string: { username: string }) =>
          showMessage(string.username)
        }
        onFinishFailed={onFinishFailed}
        saveUser={(string: { username: string }) => saveUser(string.username)}
      />
    );

    expect(await axe(container)).toHaveNoViolations();
  });

  it("renders without crashing", () => {
    const { container } = render(
      <LoginFormDesktop
        showMessage={(string: { username: string }) =>
          showMessage(string.username)
        }
        onFinishFailed={onFinishFailed}
        saveUser={(string: { username: string }) => saveUser(string.username)}
      />
    );
    expect(container).toBeTruthy();
  });

  it("matches snapshot", () => {
    const { container } = render(
      <LoginFormDesktop
        showMessage={(string: { username: string }) =>
          showMessage(string.username)
        }
        onFinishFailed={onFinishFailed}
        saveUser={(string: { username: string }) => saveUser(string.username)}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
