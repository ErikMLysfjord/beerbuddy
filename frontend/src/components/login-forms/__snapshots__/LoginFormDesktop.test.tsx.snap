// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoginFormDesktop > matches snapshot 1`] = `
<div>
  <main
    class="_container_b973b1"
  >
    <header
      class="_logoContainer_b973b1"
    >
      <a
        class="_wrapper_4f3c83"
        href="/"
      >
        <img
          alt="HopsUp logo"
          class="_icon_4f3c83"
          src="/project2/src/assets/hopsup.svg"
        />
        <h2
          class="_heading_4f3c83"
        >
          HopsUp
        </h2>
      </a>
    </header>
    <div
      class="_loginContainer_b973b1"
    >
      <section
        class="_loginText_b973b1"
      >
        <h2>
          Taste, Rate, Repeat -
          <br />
           
          The craft beer journey
        </h2>
        <p>
          Join the RateMyBrew community to unlock exclusive
          <br />
          features and be part of a global craft beer conversation.
        </p>
      </section>
      <section
        class="_loginFormContainer_b973b1"
      >
        <div
          class="ant-card ant-card-bordered css-dev-only-do-not-override-pr0fja"
          style="height: 100%; background-color: rgb(230, 244, 255);"
        >
          <div
            class="ant-card-body"
            style="height: 100%;"
          >
            <h1>
              Log in
            </h1>
            <form
              autocomplete="off"
              class="ant-form ant-form-vertical css-dev-only-do-not-override-pr0fja _loginForm_b973b1"
              id="basic"
            >
              <div
                class="ant-form-item css-dev-only-do-not-override-pr0fja"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-pr0fja"
                >
                  <div
                    class="ant-col ant-form-item-label css-dev-only-do-not-override-pr0fja"
                  >
                    <label
                      class="ant-form-item-required"
                      for="basic_username"
                      title="Username"
                    >
                      Username
                    </label>
                  </div>
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-pr0fja"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <input
                          aria-required="true"
                          class="ant-input css-dev-only-do-not-override-pr0fja"
                          id="basic_username"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="ant-form-item css-dev-only-do-not-override-pr0fja"
              >
                <div
                  class="ant-row ant-form-item-row css-dev-only-do-not-override-pr0fja"
                >
                  <div
                    class="ant-col ant-form-item-control css-dev-only-do-not-override-pr0fja"
                  >
                    <div
                      class="ant-form-item-control-input"
                    >
                      <div
                        class="ant-form-item-control-input-content"
                      >
                        <button
                          class="ant-btn css-dev-only-do-not-override-pr0fja ant-btn-primary _loginButton_b973b1"
                          type="submit"
                        >
                          <span>
                            Submit
                          </span>
                        </button>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </form>
          </div>
        </div>
      </section>
    </div>
  </main>
</div>
`;
